### ISSUE TITLE
New resource: Transit Gateway Route Table

### ISSUE THREAD

**Issue by ajkerrigan** [cloud-custodian/cloud-custodian#8253]
### Describe the feature

Custodian currently exposes Transit Gateway details via the `aws.transit-gateway` resource, and we can see TGW references in route tables through the `route` filter of the `aws.route-table` resource. But we _don't_ expose TGW route table details. The core of that comes through this call:

[describe_transit_gateway_route_tables](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html?highlight=transit_#EC2.Client.describe_transit_gateway_route_tables)

If we're going to expose route tables, we probably also want to show associations and propagations. My gut says that would be part of an augment or resource-specific filter, but the relevant calls are:

- [get_transit_gateway_route_table_propagations](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html?highlight=get_transit#EC2.Client.get_transit_gateway_route_table_propagations)
- [get_transit_gateway_route_table_associations](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html?highlight=get_transit#EC2.Client.get_transit_gateway_route_table_associations)

**Comment by kapilt** [comment]
the transit gateway route table sounds good, but it would be good to have a more concrete use case on a policy goal for it or associated filters.

**Comment by calebsyring** [comment]
I've got this one

**Comment by calebsyring** [comment]
@ajkerrigan basic implementation looks like this:

```python
@resources.register('transit-gateway-route-table')
class TransitGatewayRouteTable(query.QueryResourceManager):

    class resource_type(query.TypeInfo):
        id = name = 'TransitGatewayRouteTableId'
        service = 'ec2'
        enum_spec = ('describe_transit_gateway_route_tables', 'TransitGatewayRouteTables', None)

        arn_type = 'transit-gateway-route-table'
        date = 'CreationTime'
        cfn_type = config_type = 'AWS::EC2::TransitGatewayRouteTable'
        id_prefix = 'tgw-rtb-'

    def augment(self, resources):
        client = local_session(self.session_factory).client('ec2')

        for r in resources:
            propagations = client.get_transit_gateway_route_table_propagations(
                TransitGatewayRouteTableId=r['TransitGatewayRouteTableId'])
            r['Propagations'] = propagations['TransitGatewayRouteTablePropagations']

            associations = client.get_transit_gateway_route_table_associations(
                TransitGatewayRouteTableId=r['TransitGatewayRouteTableId'])
            r['Associations'] = associations['Associations']

        return resources
```

A have a couple of questions:
- Is this sufficient for the desired use case? Are there any particular filters that you would like me to make sure are covered in the tests?
- This does have an n+1 issue with API calls. Would the preference here be to query all propagations and associations in the account and then match on id instead?

**Comment by ajkerrigan** [comment]
>     Is this sufficient for the desired use case? Are there any particular filters that you would like me to make sure are covered in the tests?

Yes I think that's sufficient. The initial requested use case I relayed here was for querying this data and correlating it outside the scope of Custodian. I could see there being some use for a related resource style filter looking for VPCs that have TGW associations or propagations, but the folks I've spoken with haven't expressed a need for that at this point.

No particular extra filters at this time.

>    This does have an n+1 issue with API calls. Would the preference here be to query all propagations and associations in the account and then match on id instead?

It's nice to make bulk calls when we can, but I don't see a way to do that for these calls. Your sample code looks like the best we can do. The main alternatives I was considering:

- Use a dedicated filter rather than an augment?
  - ...but the resource doesn't feel useful without this info.
- Move associations & propagations to child resources of a parent route table resource?
  - ...but that seems more complex and decoupled than actually valuable.
